### PR TITLE
ShowResponse

### PR DESCRIPTION
**Pull request by Gamemackerel**
- Add ShowResponse Class to replace the hash that returns from ShowBlocks
- Add spec test for ShowResponse
- Remove spec test files for classes that no longer seem to exist
- Fix Collection.new_collection to bind the fresh collection to a locator, like Item.make does

### CODE REVIEW

File: lib/krill/show_response.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      if (opts[:op] && opts[:row]) || (!opts[:op] && !opts[:row])
+        raise TableCellUndefined, "Invalid parameters for get_table_response - specify one of op or row, not both"
+      elsif self[:table_inputs].nil?
+        return nil
+      elsif opts[:op]
+        if self[:table_inputs].first[:opid] < 0
+          raise TableCellUndefined, "Invalid parameters for get_table_response - an :op option cannot be specified for a table that doesn't have operations corresponding to its rows"
+        end
+        opid = Operation.find(opts[:op]).id # return op.id if passed an operation or the id itself
+        target_table_input = self[:table_inputs].find { |ti| (ti[:key].to_sym == var.to_sym) && (ti[:opid] == opid) }
```
Comment (bjkeller) [inline lib/krill/show_response.rb]: Factor out lines 38 and 40 by determining whether you are using `:op` or `:row` and setting an index variable

File: lib/krill/show_response.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    # a key associated with a table response-set, returns a list of table responses
+    # in order of the rows of the table.
+    #
+    # @param var [Symbol/String]  The var used to store data under
+    #               specified in the get or select call of the ShowBlock
+    def get_response var
+      responses[var.to_sym]
+    end
+
+    # Returns data recorded in a specified row of an input table
```
Comment (bjkeller) [inline lib/krill/show_response.rb]: For all block comments, put a commented blank line between the description and the parameters -- like in the one above.

File: lib/krill/show_response.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
+module Krill
+
+  # @api krill
+  require 'delegate'
+  class ShowResponse < SimpleDelegator
```
Comment (bjkeller) [inline lib/krill/show_response.rb]: Add a comment describing this class.

File: spec/krill/show_response_spec.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# require_relative 'show_response'
+require 'rails_helper'
+require '/aquarium/lib/krill/show_response'
+include Krill
+# require_relative '../models/operation'
+
+RSpec.describe ShowResponse do
+	resp = ShowResponse.new({
+		table_inputs: [{key: "tblrespnskey", opid: 3075, row: 0, value: "2", type: "number"}, {key: "tblrespnskey", opid: 3076, row: 1, value: "1", type: "number"}, ],
```
Comment (bjkeller) [inline spec/krill/show_response_spec.rb]: format so lines are <80 characters

**Comment by Gamemackerel** [comment]
The guarantee that existing protocols will not break from the ShowBlock object being returned from `show` instead of a Hash lies in the guarantee of the SimpleDelegator class. 

From ruby-doc:
"This class provides the means to delegate all supported method calls to the object passed into the constructor"
all methods supported by the object will be delegated
This update does not change how the show return hash is generated, it only wraps it with additional methods right before it is returned by passing it into the ShowResponse Delegator class. Thus we know that the returned object will function exactly like the old show return hash normally would when accessed as a hash, but with additional methods decorating it (which do not overlap with existing hash methods).

**Pull request merged**